src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java

```diff
@@ -71,13 +73,17 @@
 
   private static final String ATTRIBUTES_SUPPORTED_GRADLE_VERSION = "4.0";
 
-  public Set<ResolvedDependency> findResolvedDependencies(Project rootProject, boolean allConfigurations) {
-    return new LinkedHashSet<>(rootProject.getAllprojects()).stream()
-        .flatMap(project -> new LinkedHashSet<>(project.getConfigurations()).stream())
+  public Set<ResolvedDependency> findResolvedDependencies(Project project, boolean allConfigurations) {
+    return new LinkedHashSet<>(new LinkedHashSet<>(project.getConfigurations()).stream()
         .filter(configuration -> isAcceptableConfiguration(configuration, allConfigurations))
-        .flatMap(configuration -> getDependencies(rootProject, configuration,
+        .flatMap(configuration -> getDependencies(project, configuration,
             resolvedConfiguration -> resolvedConfiguration.getFirstLevelModuleDependencies().stream()))
-        .collect(Collectors.toCollection(LinkedHashSet::new));
+        .collect(Collectors.toMap(ResolvedDependency::getName, Function.identity(), (existing, replacement) -> {
+          if (StringUtils.containsAny(replacement.getConfiguration().toLowerCase(Locale.ROOT), "runtime", "release")) {
+            return replacement;
+          }
+          return existing;
+        }, LinkedHashMap::new)).values());
   }
```

Comment:
Persona preference, this block of code is huge, so maybe we can split into smaller parts? What about the collectors part, creating a method for it

Reply:
> Persona preference, this block of code is huge, so maybe we can split into smaller parts? What about the collectors part, creating a method for it

Done: 6728e6f38ec379945b7790d2f4e2cd2e591c13f9